src/panels/config/logs/system-log-card.ts

```diff
@@ -52,21 +53,23 @@ export class SystemLogCard extends LitElement {
       ? this._items.map((item) => getLoggedErrorIntegration(item))
       : [];
     const filteredItems = this._items
-      ? this._items.filter((item) => {
-          if (this.filter) {
-            const filter = this.filter.toLowerCase();
-            return (
-              item.message.some((message) =>
-                message.toLowerCase().includes(filter)
-              ) ||
-              item.source[0].toLowerCase().includes(filter) ||
-              item.name.toLowerCase().includes(filter) ||
-              this._timestamp(item).toLowerCase().includes(filter) ||
-              this._multipleMessages(item).toLowerCase().includes(filter)
-            );
-          }
-          return item;
-        })
+      ? this._items.filter(
+          memoizeOne((item) => {
```

Comment:
The memoize should be higher, and take `this._items` and `this.filter` as arguments, so if one of those change, it will recalculate, otherwise return the previous result.

```ts
private _getFilteredItems = memoizeOne((items, filter) => items.filter(...));
const filteredItems = this._getFilteredItems(this._items, this.filter);
```